File: models/Gpt-3.5-turbo/cpp/code/problems/sorting_and_searching/Movie_Festival_II.cpp

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

struct Movie {
    int start;
    int end;
};

bool sortByEnd(const Movie& a, const Movie& b) {
    return a.end < b.end;
}

int main() {
    int n, k;
    cin >> n >> k;

    vector<Movie> movies(n);
    for (int i = 0; i < n; i++) {
        cin >> movies[i].start >> movies[i].end;
    }

    sort(movies.begin(), movies.end(), sortByEnd);

    int count = 0;
    for (int i = 0; i < n; i++) {
        if (k == 0) break;
        if (movies[i].start >= movies[i].end) continue;
        count++;
        k--;
    }

    cout << count << endl;

    return 0;
}

File: models/Gpt-3.5-turbo/py/code/sorting_and_searching/Movie_Festival_II.py

import heapq

# Step 1: Read input values
n, k = map(int, input().split())
movies = [list(map(int, input().split())) for _ in range(n)]

# Step 2: Sort movies based on end time
movies.sort(key=lambda x: x[1])

# Step 3: Initialize variables
count = 0
end_times = []

# Step 5: Iterate over each movie
for movie in movies:
    start_time, end_time = movie
    
    # Remove movies that have already ended
    while end_times and end_times[0] <= start_time:
        heapq.heappop(end_times)
    
    # If there are still available slots, increment count and add current movie's end time
    if len(end_times) < k:
        count += 1
        heapq.heappush(end_times, end_time)

# Step 6: Print maximum total number of movies
print(count)
